### ISSUE TITLE
Provide official docker image

### ISSUE THREAD

**Issue by tenjaa** [phrase/phrase-cli#13]
Hey :)
I already opened this request in the (old?) repo.
Would you be able to provide an official docker image?

https://github.com/phrase/phraseapp-client/issues/191

**Comment by EntePF** [comment]
Do you have any update for this? This would help us a lot and would greatly appreciate it.

**Comment by rlisowski** [comment]
@EntePF we will talk about this on next planning on Monday.

**Comment by tenjaa** [comment]
Can you tell us what you decided on that monday?
If youre willing to maintain it and just dont have time for the initial work, I think we might be able to provide a pull request

**Comment by rlisowski** [comment]
@tenjaa sorry I forgot to put an update, this is in progress, see https://github.com/phrase/phrase-cli/pull/47 for details.

**Comment by egze** [comment]
Hi @tenjaa 

Could you please give it a test and report back if anything is missing? 😃 https://hub.docker.com/r/phrase/phrase-cli/tags

`docker pull phrase/phrase-cli:2.0.15` (We will also provide a `latest` tag in future)

**Comment by dsimon-onventis** [comment]
@egze I was also waiting for this, finally it's there 🍾 

Running `phrase pull` and `phrase push` seems to work fine for me. However, `phrase init` causes an endless loop.

This is the command I ran:

```
docker run --rm phrase/phrase-cli:2.0.15 init --access_token myToken
```

Results in: 

```
Please enter your API access token (you can generate one in your profile at phrase.com): Please enter your API access token (you can generate one in your profile at phrase.com): ...
```

It would be very handy if you could run the CLI from within docker without having to install the CLI on your system.

**Comment by egze** [comment]
@dsimon-onventis We will look into it. Thanks for testing.

**Comment by tenjaa** [comment]
@egze Looks good thanks!
Our usecase would be to use it in a CI environment (Concourse) to do some stuff. There we would need jq (to format `phrase locales list`) and git (to create a PR with the changes).

I can of course understand if you say, that you dont want to install these utilities by default because then more and more requests will arise. My opinion on this topic would probably be that it is no runtime artifact but a utility. Therefore minimal size isnt that important.

Having an up2date image where we could extend from is already helping a lot, though.

**Comment by dsimon-onventis** [comment]
@tenjaa @egze I don't think that it makes sense to include anything else. IMO this is meant to be a base image which only includes Phrase and nothing additional.

**Comment by egze** [comment]
Yes, we also thought about it and decided to only include the minimum. You could then use our image as a base image and add other packages that you need.

Would that be OK for you @tenjaa ?

**Comment by tenjaa** [comment]
Sure no problem!

**Comment by LarsLueecke** [comment]
Hi. Can you provide a latest tag, so we don't need to specify the concrete version?

**Comment by egze** [comment]
> Running `phrase pull` and `phrase push` seems to work fine for me. However, `phrase init` causes an endless loop.

`init` is interactive and there is no mode without interaction. The correct usage would be: `docker run -it phrase/phrase-cli:2.0.15 init …`

Please let me know if this gets you further.

**Comment by dsimon-onventis** [comment]
@egze With the `-it` flag it works flawlessly. Thanks for pointing out!

Maybe you can add a section about running the CLI via Docker somewhere in the documentation. I think it is a great new feature that you can run the CLI now with only Docker having installed!

**Comment by egze** [comment]
Done https://github.com/phrase/phrase-cli#5-docker

Thanks for your feedback!
